azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -520,56 +520,58 @@ private void handleDownloadProject(HttpServletRequest req,
 
   }
 
-    /**
-     * validate readiness of a project and user permission and use
-     * projectManager to purge the project if things looks good
-     **/
-    private void handlePurgeProject(HttpServletRequest req,
-        HttpServletResponse resp, Session session) throws ServletException,
-        IOException {
-        User user = session.getUser();
-        String projectName = getParam(req, "project");
-        HashMap<String, Object> ret = new HashMap<String, Object>();
-        boolean isOperationSuccessful = true;
-
-        if (projectManager.isActiveProject(projectName)) {
-            ret.put("error", "Project " + projectName
-                + " should be deleted before purging");
-            isOperationSuccessful = false;
-        }
+  /**
+   * validate readiness of a project and user permission and use projectManager to purge the project
+   * if things looks good
+   **/
+  private void handlePurgeProject(HttpServletRequest req, HttpServletResponse resp, Session session)
+          throws ServletException, IOException {
+    User user = session.getUser();
+    HashMap<String, Object> ret = new HashMap<String, Object>();
+    boolean isOperationSuccessful = true;
 
-        try {
-            Project project = null;
+    try {
+      Project project = null;
 
-            // project is already deleted
-            if (isOperationSuccessful) {
-                project = projectManager.getProject(projectName);
-                if (project == null) {
-                    ret.put("error", "no project with name : " + projectName);
-                    isOperationSuccessful = false;
-                }
-            }
+      if (hasParam(req, "id")) {
+        project = projectManager.getProject(getParam(req, "id"));
```

Comment:
you need to cast the getParam result to int here , otherwise getProject will regard it as name but id .